src/test/java/automationpractice/com/pages/ProductDetailsPage.java

```diff
@@ -0,0 +1,251 @@
+package automationpractice.com.pages;
+
+import net.serenitybdd.core.pages.PageObject;
+import net.serenitybdd.core.pages.WebElementFacade;
+import org.openqa.selenium.support.FindBy;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+public class ProductDetailsPage extends PageObject {
+
+    @FindBy(id = "add_to_cart")
+    private WebElementFacade clickOnAddToCartButton;
+
+    @FindBy(xpath = "//*[contains(@class,'button-medium')]//*[contains(@class,'icon-chevron-right')]")
+    private WebElementFacade clickProceedToCheckoutButton;
+
+    @FindBy(className = "icon-plus")
+    private WebElementFacade clickPlusQuantityButtonOnProductDetailsPage;
+
+    @FindBy(className = "icon-minus")
+    private WebElementFacade clickMinusQuantityButtonOnProductDetailsPage;
+
+    @FindBy(xpath = "//*[@id='group_1']//option[1]")
+    private WebElementFacade selectSSize;
+
+    @FindBy(xpath = "//*[@id='group_1']//option[2]")
+    private WebElementFacade selectMSize;
+
+    @FindBy(xpath = "//*[@id='group_1']//option[3]")
+    private WebElementFacade selectLSize;
+
+    @FindBy(id = "color_14")
+    private WebElementFacade changeColorToBlue;
+
+    @FindBy(id = "color_13")
+    private WebElementFacade changeColorToOrange;
+
+    @FindBy(id = "color_8")
+    private WebElementFacade changeColorToWhite;
+
+    @FindBy(id = "color_11")
+    private WebElementFacade changeColorToBlack;
+
+    @FindBy(id = "color_15")
+    private WebElementFacade changeColorToGreen;
+
+    @FindBy(id = "color_16")
+    private WebElementFacade changeColorToYellow;
+
```

Comment:
Podobnie tutaj. Jeśli idiki kolorów się zmienią to ten test nie przejdzie. Plus tworzycz 6 xpath gdzie można stworzyć jednego Stinga //*[@id="color_to_pick_list"]//*[@title='%s'] i tylko podmieniać %s